src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ExpressionMetadataProvider.cs

```diff
@@ -50,6 +50,14 @@ public static class ExpressionMetadataProvider
                     // Property/field access is always legal
                     var memberExpression = (MemberExpression)expression.Body;
                     propertyName = memberExpression.Member is PropertyInfo ? memberExpression.Member.Name : null;
+                    if (string.Equals(propertyName, "Model", StringComparison.Ordinal) &&
```

Comment:
Kind of a derivative case but possible. I guess this code could check whether the LHS is a `RazorPage<TModel>` instance. Lemme see how ugly that gets.

If clean, would break the `m => ViewData.Model` support we get for free right now. But `ExpressionHelper` gets the names wrong for that one anyway. No biggie either way.